<?xml version="1.0"?>
<launch>

    <arg name="start_pt" default= '-2.00, 3.60, -2.2, 0'/>
    <arg name="use_rviz" default="true"/>
    <arg name="use_matplotlib" default="false"/>
    <arg name="is_sim" default="true"/>
    <arg name="map_file" default="$(find svea_core)/resources/maps/q1.yaml" />
    <arg name="map_name" default="q1.yaml" />

    <!--
    <group if="$(eval use_rviz == True)">
        <node type="rviz" name="rviz" pkg="rviz" args="-d $(find svea_core)/resources/rviz/SVEA_floor2.rviz" />
    </group>
    -->
    <!--open serial connection for controlling SVEA-->
	<group if="$(eval is_sim == true)">
        <node pkg="map_server" type="map_server" name="map_server" output="screen" args="$(arg map_file)" />
        <node type="rviz" name="rviz" pkg="rviz" args="-d $(find svea_core)/resources/rviz/SVEA_floor2.rviz" />
    </group>
    
    <node pkg="svea_core" type="path_planner_q1.py" name="path_planner" output="screen" />
    <node pkg="svea_core" type="emergency_break.py" name="emergency_break" output="screen" />
    <node pkg="svea_core" type="state_to_pixels_map.py" name="state_to_pixels_map" output="screen" />
    <node pkg="svea_core" type="map_logic.py" name="map_logic" output="screen" />
    <node pkg="svea_core" type="speed_steering.py" name="speed_steering" output="screen" />
    <node pkg="svea_core" type="control_filter.py" name="control_filter" output="screen" />
    <node pkg="svea_core" type="republisher.py" name="republisher" output="screen" />
    <node pkg="svea_core" type="A_star.py" name="A_star" output="screen" />

    
    <node pkg="svea_core" type="q1_example.py" name="q1_example" output="screen">
        <param name="start_pt" value="$(arg start_pt)"/>
        <param name="use_rviz" value="$(arg use_rviz)"/>
        <param name="use_matplotlib" value="$(arg use_matplotlib)"/>
        <param name="is_sim" value="$(arg is_sim)"/>
    </node>
    
    <rosparam command="load" file="$(find svea_core)/resources/params/q1obs.yaml" />
    <rosparam command="load" file="$(find svea_core)/resources/params/race_track.yaml" />
    <rosparam command="load" file="$(arg map_file)" />

</launch>
